packages/marshal/test/test-marshal-capdata.js

```diff
@@ -383,6 +384,43 @@ test('records', t => {
   shouldThrow(['nonenumStringData', 'enumStringData'], REC_ONLYENUM);
 });
 
+test('promises', t => {
+  const { serialize, unserialize } = makeMarshal(
+    val => val,
+    slot => slot,
+    {
+      serializeBodyFormat: 'capdata',
+    },
+  );
+  const assertRoundTrip = (val, body, slots, description) => {
+    t.deepEqual(
+      serialize(val),
+      { body, slots },
+      `${description} can be serialized`,
+    );
+    const val2 = unserialize(harden({ body, slots }));
+    t.deepEqual(val, val2, `${description} round-trips`);
+  };
+
+  const p = harden(Promise.resolve(null));
+  assertRoundTrip(p, `{"@qclass":"slot","index":0}`, [p], 'Promise');
+  const fakeP = harden(
+    Object.defineProperties(
+      {},
+      {
+        [PASS_STYLE]: { value: 'promise' },
+        [Symbol.toStringTag]: { value: 'Pseudo-promise' },
+      },
+    ),
+  );
+  assertRoundTrip(
+    fakeP,
+    `{"@qclass":"slot","index":0}`,
+    [fakeP],
+    'ersatz promise',
```

Comment:
This PR generally calls these 'pseudo-promises', including immediately above, but also immediately above suggests that they are "fake" and "ersatz".

I'm not sure what to do about this terminology problem. According to `passStyleOf` and therefore our distributed objects system, there are promises. When passed across a marshal barrier, they unserialize to genuine promises. Thus, from the perspective of the distributed object system, their promise nature is pass invariant. This is all as it should be.

Normally I'd suggest perhaps "virtual promise", but we should now use that only for the ones that can be paged out. I'm ok with "pseudo", but we should not additionally imply that they are fake or esratz.

Reply:
Also elsewhere

Reply:
> When passed across a marshal barrier, they unserialize to genuine promises.

They may not.


> I'm ok with "pseudo", but we should not additionally imply that they are fake or esratz.

I'm also ok with pseudo, but would be happy to find another term that can later be exposed in types (since the non native promise will be visible to user-land)

Reply:
I went through a bit of churn before settling on "pseudo-promise", and remain open to alternatives. But in the meantime, I've replaced references to "ersatz promise", although variables in test files still use a "fake" prefix because "pseudo" sucks to type.